src/components/android-settings-container.js

```diff
@@ -74,12 +85,20 @@ class AndroidSettingsContainer extends React.Component {
             return true
         }
 
+        let { numberSmsToSend, numberSmsConfirmation } = this.state
+        numberSmsConfirmation == ''
+            ? (numberSmsConfirmation = null)
+            : (numberSmsConfirmation = numberSmsConfirmation)
```

Comment:
This should be written the other way around if using a ternary operator (`a = a === '' ? b : a`), but here the re-assignment only happens in one condition so it's better and easier-to-read like this:

```js
if (this.state.numberSmsConfirmation === '') {
    this.state.numberSmsConfirmation = null
}
```